Update best practices (#27)
finn-no/ios-handbook: BEST_PRACTICES.md

# Dos and don'ts

> ## SDKs tag versioning

Reviewer: ```suggestion
## Libraries/SDKs versioning

### MAJOR.MINOR.PATCH
When making a new release/version keep in mind how to update the version number.
- Major: The change you are making is a breaking change, meaning that "users" of it can't update without also making code changes.
- Minor: You added something new or updated something but it does not require code changes.
- Patch: Fix for a bug or error in a way that is internal to the library/SDK.
```

> ### What?

Reviewer: ```suggestion
## Working process guide
```

After merging any PR to any SDK's master, please don't forget to make a new release (at least patch) using `bundle exec fastlane` script.

> After creating a new SKD's tag please bump it's version on ios-app. If you have any blockers to do so - please make a note inside PR you closed.

Reviewer 2: Typo: SKD -> SDK

> After merging any PR to any SDK's master, please don't forget to make a new release (at least patch) using `bundle exec fastlane` script.
> After creating a new SDK's tag please bump it's version on ios-app. If you have any blockers to do so - please make a note inside PR you closed.

Reviewer: ```suggestion
After merging a PR to master, please don't forget to make a new release (at least patch). There should be a script to do this (`bundle exec fastlane`), never update version numbers inside a PR.

#### Breaking changes
When you do a breaking change the "users" of that repo should be updated ASAP. The recommended approach is to open PRs on those repositories as well and wait for all of them to be approved before merging and making a new major release.

#### Minor/patch changes 
If you will be following up with more PRs you can wait and make a new minor/patch release after they are all merged, provided you know that no one else will make changes in the meanwhile. Do not forget to update ios-app (or other "owners") after making the new version. If you will/can not do this directly leave a note of it in the PR you closed.
```
